src/system-info.ts

```
@@ -0,0 +1,154 @@
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      this.observer = new PressureObserver(this.handleStateChange.bind(this));
+
+      this.observer.observe('cpu');
+    }
+  }
+
+  /**
+   * Handles updates from the PressureObserver.
+   *
+   * @param records - The records from the PressureObserver.
+   */
+  private handleStateChange(records: PressureRecord[]) {
+    records.forEach((record: PressureRecord) => {
+      if (record.source === 'cpu' && record.state !== this.lastCpuPressure) {
+        this.lastCpuPressure = record.state;
+
+        this.emit(SystemInfoEvents.CpuPressureStateChange, record.state);
+      }
+    });
+  }
+
+  /**
+   * Gets the current CPU pressure state.
+   *
+   * @returns The current CPU pressure state, or undefined if API is not supported.
+   */
+  getCpuPressure(): PressureState | undefined {
+    return this.lastCpuPressure;
+  }
+
+  /**
+   * Checks if the Compute Pressure API is supported in the current environment.
+   *
+   * @returns True if the Compute Pressure API is supported, false otherwise.
+   */
+  static isPressureObserverSupported(): boolean {
+    return 'PressureObserver' in window;
+  }
+}
+
+const systemInfo = new SystemInfoInternal();
+
+/**
+ * SystemInfo class to provide static methods for system information.
+ */
+export class SystemInfo {
+  /**
+   * Checks if the Compute Pressure API is supported in the current environment.
+   *
+   * @returns True if the Compute Pressure API is supported, false otherwise.
+   */
+  static isPressureObserverSupported(): boolean {
+    return SystemInfoInternal.isPressureObserverSupported();
+  }
+
+  /**
+   * Gets the current CPU pressure state.
+   *
+   * @returns The current CPU pressure state, or undefined if API is not supported.
+   */
+  static getCpuPressure(): PressureState | undefined {
+    if (!SystemInfo.isPressureObserverSupported()) {
+      return undefined;
+    }
```

Do we want to be more strict here and have as well throw new `Error`? Are there any cases where we need to keep it `undefined`? Just feels like we have `lastCpuPressure` as `undefined` and then how to understand if `getCpuPressure` supports or not?

---

I guess we could just return here. But from user perspective, it would be great to have a way to know if listener was applied or not.